tests/application/cms/test_page_service.py

```diff
@@ -428,6 +428,25 @@ def test_create_new_major_version_does_not_duplicate_uploads(self):
         assert len(measure_version.uploads) == 1
         assert len(new_version.uploads) == 0
 
+    @pytest.mark.parametrize("version_type", [t for t in NewVersionType])
+    def test_create_measure_version_creates_associations_rather_than_copies_of_existing_data_sources(
+        self, version_type
+    ):
+        measure_version = MeasureVersionFactory(version="1.0")
+        user = UserFactory(user_type=TypeOfUser.RDU_USER)
+
+        assert len(measure_version.data_sources) == 1
+        assert DataSource.query.count() == 1
+
+        new_version = page_service.create_measure_version(measure_version, version_type, user=user)
+
+        assert DataSource.query.count() == 1
+
+        if version_type == NewVersionType.MAJOR_UPDATE:
```

Comment:
Feels a bit weird to have an `if` condition in a test assertion. Possibly it'd be more understandable (if verbose) to have separate tests for major updates vs minor updates?

Reply:
Sure 👍